src/ThreeEditor/js/Viewport.ClipPlane.js

```diff
@@ -0,0 +1,165 @@
+import * as THREE from 'three';
+import { GUI } from 'three/examples//jsm/libs/dat.gui.module.js';
+
+export function ViewportClippedView(editor, viewport, planeHelpers, initialObjects, signalGeometryChanged, { clipPlane, planeHelperColor, planePosLabel }) {
+
+    const STENCIL_RENDER_ORDER = 1;
+    const CLIPPING_SIZE = 10;
+
+    this.scene = new THREE.Scene();
+    this.gui = new GUI({});
+
+    // Setup plane clipping ui  
+    const planeHelper = new THREE.PlaneHelper(clipPlane, CLIPPING_SIZE, planeHelperColor ?? 0xffff00); // default helper color is yellow
+    planeHelpers.add(planeHelper);
+
+    const planePosProperty = planePosLabel ?? 'PlanePos';
+
+    const uiProps = {
+
+        get [planePosProperty]() {
+
+            return clipPlane.constant;
+
+        },
+        set [planePosProperty](v) {
+
+            crossSectionPlane.translateZ(v - clipPlane.constant); // translate Cross Section Plane with clip plane
+
+            clipPlane.constant = v;
+
+            editor.signals.viewportConfigChanged.dispatch({ name: viewport.name, globalPlaneConstant: v });
+
+        },
+
+        get 'Helper Visible'() {
+
+            return planeHelper.visible;
+
+        },
+        set 'Helper Visible'(v) {
+
+            planeHelper.visible = v;
+
+            editor.signals.viewportConfigChanged.dispatch({ name: viewport.name, helperVisible: v });
+
+        },
+
+
+    };
+
+    this.gui.add(uiProps, planePosProperty, -CLIPPING_SIZE, CLIPPING_SIZE, 0.1);
```

Comment:
What `add` method is doing?
What is the meaning of following arguments `-CLIPPING_SIZE, CLIPPING_SIZE, 0.1` ?
Does it mean that something is spanning from -10 to 10 ? If that is the case, then the name is misleading, as size of something that spans from -10 to +10 is 20 (in such case 10 is rather a half-length).